auction-app-frontend/src/forms/rules/validationRules.js

```diff
@@ -37,5 +43,95 @@ export const rules = {
             /^(?=.*[A-Z])(?=.*[a-z])(?=.*[0-9]).{8,}$/, 
             "Password must contain at least one uppercase letter, one lowercase letter, and one number"
         )
+    }), 
+    creditCard: () => ({
+        ...rules.required("Credit Card"),
+        ...rules.pattern(/^\d{16}$/, "Enter a valid 16-digit number"),
+        validate: (value) => checkLuhn(value) || "Enter a valid credit card number"
+    }),
+    expirationYear: () => ({
+        ...rules.required("Year"),
+        ...rules.pattern(/^\d{2}$/, "Enter a valid year (YY)"),
+        validate: (value) => validateYear(value) || "Year cannot be in the past"
+    }),
+    expirationMonth: () => ({
+        ...rules.required("Month"),
+        ...rules.pattern(/^(0[1-9]|1[0-2])$/, "Enter a valid month (MM)")
+    }),
+    cvv: () => ({
+        ...rules.required("CVV"),
+        ...rules.minLength(3, "CVV"),
+        ...rules.maxLength(4, "CVV"),
+        ...rules.pattern(/^\d{3,4}$/, "CVV must be 3 or 4 digits")
+    }),        
+    street: () => ({
+        ...rules.required("Street"),
+        ...rules.minLength(3, "Street"),
+        ...rules.maxLength(100, "Street"),
+        ...rules.pattern(/^[a-zA-Z0-9\s,."-]{3,100}$/, "Invalid street name")
+    }),
+    city: () => ({
+        ...rules.required("City"),
+        ...rules.minLength(2, "City"),
+        ...rules.maxLength(50, "City"),
+        ...rules.pattern(/^[a-zA-Z\s]{2,50}$/, "Invalid city name")
+    }),
+    state: () => ({
+        ...rules.required("State"),
+        ...rules.minLength(2, "State"),
+        ...rules.maxLength(50, "State"),
+        ...rules.pattern(/^[a-zA-Z\s]{2,50}$/, "Invalid state name")
+    }),
+    country: () => ({
+        ...rules.required("Country"),
+        ...rules.minLength(2, "Country"),
+        ...rules.maxLength(50, "Country"),
+        ...rules.pattern(/^[a-zA-Z\s]{2,50}$/, "Invalid country name")
+    }),
+    month: () => ({
+        ...rules.required("Month"),
+        ...rules.minValue(1, "Month"),
+        ...rules.maxValue(12, "Month"),
+        ...rules.pattern(/^[0-9]{1,2}$/, "Invalid month")
+    }),
+    day: () => ({
+        ...rules.required("Day"),
+        ...rules.minValue(1, "Day"),
+        ...rules.maxValue(31, "Day"),
+        ...rules.pattern(/^[0-9]{1,2}$/, "Invalid day")
+    }),
+    year: () => ({
+        ...rules.required("Year"),
+        ...rules.minValue(1930, "Year"),
+        ...rules.maxValue(2024, "Year"),
+        ...rules.pattern(/^[0-9]{2,4}$/, "Invalid year")
     })
 };
+
+const checkLuhn = (cardNo) => {
```

Comment:
If you use some code from the internet, please just adapt it with our codebase. 
Don't use short names or name of the alogrithms. Use full names and specify why you need it 
Instead of checkLuhn can be checkCardNumberValidity